src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.CSharp.DesignTime.targets

```
@@ -64,6 +64,39 @@
 
   <!-- Targets -->
 
+<!-- Razor cohosting support for projects that don't use the Razor SDK -->
+
+  <!--
+    If there are any .razor or .cshtml files in this project we want to do three things:
+
+      1. Add a 'RazorWithoutSdk' capability to the project so tooling can light up as necessary
+      2. Reference the Razor source generator
+      3. Add each .razor and .cshtml file to the project as an AdditionalFiles item so the source generator can see it
+  -->
+
+  <!-- .NET Framework uses a different editor in VS -->
+ <Target Name="DetectRazorOrCshtmlFiles" BeforeTargets="AssignProjectConfiguration" Condition="'$(DesignTimeBuild)' == 'true' And '$(TargetFrameworkIdentifier)' != '.NETFramework' And '$(UsingMicrosoftNETSdkRazor)' != 'true'">
+    <ItemGroup>
+      <!-- Collect all Razor and CSHTML files -->
+      <_RazorOrCshtmlFiles Include="**\*.razor;**\*.cshtml" />
+    </ItemGroup>
+
+    <!-- Add a capability to control initialization of Razor in the IDE -->
+    <ItemGroup Condition="'@(_RazorOrCshtmlFiles)' != ''">
+      <ProjectCapability Include="RazorWithoutSdk" />
+    </ItemGroup>
+
+    <!-- As a get out of jail card, we let users turn off anything impactful, but their IDE won't really work. -->
+    <ItemGroup Condition="'$(DisableAllRazorFilesAsAdditionalFiles)' != 'true' And '@(_RazorOrCshtmlFiles)' != ''">
+      <AdditionalFiles Include="@(_RazorOrCshtmlFiles)" Condition="'@(_RazorOrCshtmlFiles)' != ''" />
```

Redundant condition in this `ItemGroup` I think.

```suggestion
      <AdditionalFiles Include="@(_RazorOrCshtmlFiles)" />
```